sdk/nodejs/stackReference.ts

```diff
@@ -110,6 +118,23 @@ export class StackReference extends CustomResource {
         return promiseResult(out.promise());
     }
 
+    async isSecretOutput(name: Output<string>): Promise<boolean> {
+        // If either the name or set of secret outputs is unknown, we can't do anything smart, so we just copy the
+        // secretness from the entire outputs value.
+        if (!((await name.isKnown) && (await this.secretOutputs.isKnown))) {
```

Comment:
unless i'm mistaken you only ever call this from callers who pass in `output(name)`.  in which case (even if `name` is `undefined`) then `await name.isKnown` will always return true.

Reply:
about hte latter case, when would that happen?  are there cases were we don't 'know' the `outputs` or `secretOutputNames` of a stack resource?  Is that during first preview for example?

Reply:
I believe this is not true, but let me check my thinking.  If we look at the callers of `isSecretOutput` they both call `output(name)`, but `name` itself is an `Input<string>`.  That means it could already be an Output<string> you had gotten somewhere else.  In that case, the `output(name)` is logically a "no-op" and so the result may not still be known.

@pgavlin tells me that we do actually do `read` operations during a previews, and because StackReference is modeled as a read, I believe that `this.secretOutputs.isKnown` should always be true.  I can remove this clause, but I sort of like having it in there as a backstop.

Reply:
> I can remove this clause, but I sort of like having it in there as a backstop.

I'm on the fence.  Basically, we're close to 1.0 so being conservative makes sense.   However, i really dislike code that is hard to reason about because it should never be hit.  it makes future work in this area harder.  if we feel we have time to kick the tires, i woudl far prefer doing hte check, but asserting that it should always be isKnown.  that way if it turns out not to be we know exactly the case and can doc the code as such.

I'm def on the fence here... but given the complexity in this area, my personal pref is to err toward maintainability.

Reply:
> the output(name) is logically a "no-op" and so the result may not still be known.

You are correct.  Good call.